### PR TITLE
Add onvif mapper based kubeedge v1.17.0

### PR DESCRIPTION
**Pull request by wbc6080**
Add onvif mapper based mapper-framework in v1.17.0 of kubeedge

### CODE REVIEW

**Comment by luomengY** [comment]
@fisherxu

File: mappers/kubeedge-v1.17.0/onvif-mapper/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+func main() {
+	var err error
+	var c *config.Config
+
+	klog.InitFlags(nil)
+	defer klog.Flush()
+
+	if c, err = config.Parse(); err != nil {
+		klog.Fatal(err)
```
Comment (fisherxu) [inline mappers/kubeedge-v1.17.0/onvif-mapper/cmd/main.go:25]: The `Fatal` will also exit, ptal

File: mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/redis/client.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	pong, err := RedisCli.Ping(context.Background()).Result()
+	if err != nil {
+		klog.Errorf("init redis database failed, err = %v", err)
+		return err
+	} else {
+		klog.V(1).Infof("init redis database successfully, with return cmd %s", pong)
+	}
+	return nil
+}
+
```
Comment (fisherxu) [inline mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/redis/client.go:59]:
```suggestion
	if err != nil {
		klog.Errorf("init redis database failed, err = %v", err)
		return err
	} 
	klog.V(1).Infof("init redis database successfully, with return cmd %s", pong)
	return nil
}

```

File: mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/tdengine/client.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	password := os.Getenv("PASSWORD")
+	dsn := fmt.Sprintf("%s:%s@http(%s)/%s", username, password, d.TDEngineClientConfig.Addr, d.TDEngineClientConfig.DBName)
+	var err error
+	DB, err = sql.Open("taosRestful", dsn)
+	if err != nil {
+		klog.Errorf("init TDEngine db fail, err= %v:", err)
+	} else {
+		klog.V(1).Infof("init TDEngine database successfully")
+	}
+	return nil
```
Comment (fisherxu) [inline mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/tdengine/client.go:50]: ditto

File: mappers/kubeedge-v1.17.0/onvif-mapper/resource/secret.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  password: MTIzNDU2   # Fill in based on your camera user password, the default is 123456
```
[the remark this answers is not in the document]
Comment (wbc6080) [reply mappers/kubeedge-v1.17.0/onvif-mapper/resource/secret.yaml]: Done

File: mappers/kubeedge-v1.17.0/onvif-mapper/cmd/main.go (hunk 1 above)
Comment (wbc6080) [reply mappers/kubeedge-v1.17.0/onvif-mapper/cmd/main.go:25]: I have fixed it

File: mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/redis/client.go (hunk 2 above)
Comment (wbc6080) [reply mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/redis/client.go]: Done

File: mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/tdengine/client.go (hunk 3 above)
Comment (wbc6080) [reply mappers/kubeedge-v1.17.0/onvif-mapper/data/dbmethod/tdengine/client.go]: Done

**Comment by wbc6080** [comment]
The code has been modified according to the comments @fisherxu. At the same time, since part of the code is generated by mapper-framework, I think mapper-framework still needs to be modified. I will submit a PR to the main repo later.

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @luomengY

**Pull request merged**